File: app/src/main/java/com/glucocontrol/domain/model/GlucoseStatus.kt
package com.glucocontrol.domain.model

/**
 * Clasificación clínica de una lectura de glucosa respecto al rango objetivo del usuario.
 * Se calcula siempre a partir de [GlucoseRange] para garantizar coherencia visual con Historial.
 */
enum class GlucoseStatus { LOW, NORMAL, HIGH }

fun GlucoseReading.glucoseStatus(range: GlucoseRange): GlucoseStatus = when {
    valueMgDl < range.minTarget -> GlucoseStatus.LOW
    valueMgDl > range.maxTarget -> GlucoseStatus.HIGH
    else -> GlucoseStatus.NORMAL
}

File: lib/domain/model/glucose_status.dart
import 'package:glucocontrol/domain/model/glucose_range.dart';
import 'package:glucocontrol/domain/model/glucose_reading.dart';

enum GlucoseStatus { low, normal, high }

extension GlucoseStatusX on GlucoseReading {
  GlucoseStatus glucoseStatus(GlucoseRange range) {
    if (valueMgDl < range.minTarget) return GlucoseStatus.low;
    if (valueMgDl > range.maxTarget) return GlucoseStatus.high;
    return GlucoseStatus.normal;
  }
}
